File: src/types/link.ts
export interface Link{
    id : string;
    user_id: string;
    title: string;
    slug: string;
    destination_url: string;
    is_active: boolean;
    click_count: number;
    created_at : string;
    updated_at : string;
}

export interface CreateLinkRequest {
    title: string;
    slug: string;
    destination_url: string;
}

export interface UpdateLinkRequest {
    title ?: string;
    slug?: string;
    destination_url?: string;
}

export interface ToggleStatusRequest {
    is_active : boolean;
}

export interface LinkListResponse {
    links: Link[];
}

export interface LinkStatusResponse{
    message: string;
    is_active: boolean;
}

File: internal/models/link.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Link struct {
	ID     uuid.UUID `json:"id"`
	UserID uuid.UUID `json:"user_id"`

	Title          string    `json:"title"`
	Slug           string    `json:"slug"`
	DestinationURL string    `json:"destination_url"`
	IsActive       bool      `json:"is_active"`
	ClickCount     int64     `json:"click_count"`
	CreatedAt      time.Time `json:"created_at"`
	UpdatedAt      time.Time `json:"updated_at"`
}

type UpdateLinkStatusRequest struct {
	IsActive bool `json:"is_active"`
}
